pkg/spec/zz_generated.deepcopy.go

```diff
@@ -0,0 +1,348 @@
+		}
+	}
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new NatsClusterList.
+func (in *NatsClusterList) DeepCopy() *NatsClusterList {
+	if in == nil {
+		return nil
+	}
+	out := new(NatsClusterList)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyObject is an autogenerated deepcopy function, copying the receiver, creating a new runtime.Object.
+func (in *NatsClusterList) DeepCopyObject() runtime.Object {
+	if c := in.DeepCopy(); c != nil {
+		return c
+	} else {
+		return nil
+	}
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *PodPolicy) DeepCopyInto(out *PodPolicy) {
+	*out = *in
+	if in.Labels != nil {
+		in, out := &in.Labels, &out.Labels
+		*out = make(map[string]string, len(*in))
+		for key, val := range *in {
+			(*out)[key] = val
+		}
+	}
+	if in.NodeSelector != nil {
+		in, out := &in.NodeSelector, &out.NodeSelector
+		*out = make(map[string]string, len(*in))
+		for key, val := range *in {
+			(*out)[key] = val
+		}
+	}
+	in.Resources.DeepCopyInto(&out.Resources)
+	if in.Tolerations != nil {
+		in, out := &in.Tolerations, &out.Tolerations
+		*out = make([]v1.Toleration, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	if in.NatsEnv != nil {
+		in, out := &in.NatsEnv, &out.NatsEnv
+		*out = make([]v1.EnvVar, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new PodPolicy.
+func (in *PodPolicy) DeepCopy() *PodPolicy {
+	if in == nil {
+		return nil
+	}
+	out := new(PodPolicy)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *StaticTLS) DeepCopyInto(out *StaticTLS) {
+	*out = *in
+	if in.Member != nil {
+		in, out := &in.Member, &out.Member
+		if *in == nil {
+			*out = nil
+		} else {
+			*out = new(MemberSecret)
+			**out = **in
+		}
+	}
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new StaticTLS.
+func (in *StaticTLS) DeepCopy() *StaticTLS {
+	if in == nil {
+		return nil
+	}
+	out := new(StaticTLS)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *TLSPolicy) DeepCopyInto(out *TLSPolicy) {
+	*out = *in
+	if in.Static != nil {
+		in, out := &in.Static, &out.Static
+		if *in == nil {
+			*out = nil
+		} else {
+			*out = new(StaticTLS)
+			(*in).DeepCopyInto(*out)
+		}
+	}
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new TLSPolicy.
+func (in *TLSPolicy) DeepCopy() *TLSPolicy {
+	if in == nil {
+		return nil
+	}
+	out := new(TLSPolicy)
+	in.DeepCopyInto(out)
+	return out
+}
```

Comment:
That's one thing I'd like to discuss with you. ATM the build doesn't generate code automatically. We have two options:

1) Run `hack/codegen.sh` manually after each change to `pkg/spec/*` and version the generated file (the current approach).
2) Add a `RUN ./hack/codegen.sh` step in the `Dockerfile` and ignore `zz_generated.*.go`.

Which one do you prefer? I was about to go with (2), in fact, but ended up going with (1) since the _codegen_ tool is evolving quickly and I feel it's somewhat safer to version what we're actually building. But I don't have a strong opinion, so please let me know which one do you prefer.